setup.py

```diff
@@ -57,112 +57,96 @@
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "StringConverter.cpp"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "TimeConverter.cpp"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "TimeStampConverter.cpp"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "Python", "Common.cpp"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "Python", "Helpers.cpp"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "Util", "time.cpp"),
+        NANOARROW_LOGGING_SRC_DIR + "/logging.cpp",
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "nanoarrow.c"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "nanoarrow_ipc.c"),
+        os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "flatcc.c"),
+    ]
+    ext.include_dirs.append(NANOARROW_ARROW_ITERATOR_SRC_DIR)
+    ext.include_dirs.append(NANOARROW_LOGGING_SRC_DIR)
+
+    if sys.platform == "win32":
+        if not any("/std" not in s for s in ext.extra_compile_args):
+            ext.extra_compile_args.append("/std:c++17")
+    elif sys.platform == "linux" or sys.platform == "darwin":
+        if "std=" not in os.environ.get("CXXFLAGS", ""):
+            ext.extra_compile_args.append("-std=c++17")
+            ext.extra_compile_args.append("-D_GLIBCXX_USE_CXX11_ABI=0")
+        if sys.platform == "darwin" and "macosx-version-min" not in os.environ.get(
+            "CXXFLAGS", ""
+        ):
+            ext.extra_compile_args.append("-mmacosx-version-min=10.13")
+
+    # sys.platform for linux used to return with version suffix, (i.e. linux2, linux3)
+    # After version 3.3, it will always be just 'linux'
+    # https://docs.python.org/3/library/sys.html#sys.platform
+    if sys.platform == "linux":
+        ext.extra_link_args += ["-Wl,-rpath,$ORIGIN"]
+    elif sys.platform == "darwin":
+        # rpath,$ORIGIN only work on linux, did not work on darwin. use @loader_path instead
+        # fyi, https://medium.com/@donblas/fun-with-rpath-otool-and-install-name-tool-e3e41ae86172
+        ext.extra_link_args += ["-rpath", "@loader_path"]
+    return [ext]
 
-                # sys.platform for linux used to return with version suffix, (i.e. linux2, linux3)
-                # After version 3.3, it will always be just 'linux'
-                # https://docs.python.org/3/library/sys.html#sys.platform
-                if sys.platform == "linux":
-                    ext.extra_link_args += ["-Wl,-rpath,$ORIGIN"]
-                elif sys.platform == "darwin":
-                    # rpath,$ORIGIN only work on linux, did not work on darwin. use @loader_path instead
-                    # fyi, https://medium.com/@donblas/fun-with-rpath-otool-and-install-name-tool-e3e41ae86172
-                    ext.extra_link_args += ["-rpath", "@loader_path"]
 
+if _ABLE_TO_COMPILE_EXTENSIONS and not SNOWFLAKE_DISABLE_COMPILE_ARROW_EXTENSIONS:
+    extensions = get_extensions()
+
+    class MyBuildWheel(bdist_wheel):
+        def get_tag(self):
```

Comment:
```suggestion
        def get_tag(self) -> Tuple[str, str, str]:
```
This will probably need to be imported.